instrumentation-genai/opentelemetry-instrumentation-mcp/src/opentelemetry/instrumentation/mcp/instrumentation.py

```diff
@@ -0,0 +1,450 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""OpenTelemetry MCP (Model Context Protocol) instrumentation."""
+
+import json
+import logging
+from typing import Any, Callable, Collection, Coroutine, Dict, Optional, Tuple
+
+from wrapt import register_post_import_hook  # type: ignore[import-untyped]
+
+from opentelemetry import trace
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import unwrap
+from opentelemetry.propagate import get_global_textmap
+from opentelemetry.trace import SpanKind, Status, StatusCode
+
+from .attributes import MCPMethodValue, MCPSpanAttributes
+from .version import __version__
+
+_LOG = logging.getLogger(__name__)
+
+
+class McpInstrumentor(BaseInstrumentor):
+    """
+    Instrumentor for MCP (Model Context Protocol).
+
+    Provides automatic tracing for MCP client and server operations,
+    including distributed trace context propagation.
+
+    See: https://modelcontextprotocol.io/overview
+    """
+
+    # Span naming constants
+    _CLIENT_SPAN_NAME = "mcp.client"
+    _SERVER_SPAN_NAME = "mcp.server"
```

Comment:
are these defined by MCP semconv? shoud these and the attributes below be moved to `attributes.py` as well?